### DISCUSSION TITLE
OpenID Endpoint Configuration wrong urls after updating to v25

### DISCUSSION THREAD

**Discussion by Aorrihtos** [keycloak/keycloak#31104] [Q&A]
Due to some problems we’ve just encountered when trying to authenticate users in external services (Nextcloud) via Keycloak, we have just discovered something that might by related to the bug with the admin hostnames related to update v25.

When accesing to {admin_url}/realms/{realmName}/.well-known/openid-configuration, the following attributes changed it’s url after the uptate:

token_endpoint
introspection_endpoint
userinfo_endpoint
jwks_uri
registration_ednpoint
backchannel_autentication_endpoint
pushed_authorization_request_endpoint
Before the update, their urls started with the hostname-admin (previously hostname-admin-url) configured in keycloak.config, but now, they start with the hostname label. Example:

Before update → token_endpoint: [https://admin-url.es:443/realms/EXAMPLE/protocol/openid-connect/token](https://admin-url.es/realms/EXAMPLE/protocol/openid-connect/token)

After update → token_endpoint: [https://url.es:443/realms/EXAMPLE/protocol/openid-connect/token](https://url.es/realms/EXAMPLE/protocol/openid-connect/token)
This same behaviour can be noticed on the login well-known openid-configuration.

The config proxy variables before updating (hostname v1):

hostname-strict-backchannel=false
hostname-strict=true
hostname-strict-https=true
spi-sticky-session-encoder-infinispan-should-attach-route=false
proxy=passthrough
hostname-url={{ hostname_url }}
hostname-admin-url={{ hostname-admin_url }}

The config proxy after updating (hostname v2):

hostname-backchannel-dynamic=true
hostname-strict=true
spi-sticky-session-encoder-infinispan-should-attach-route=false
hostname={{ hostname_url }}
hostname-admin={{ hostname-admin_url }}

Thanks in advance.

We moved from version 24.0.5 to 25.0.1.

**Comment by Akvel** [comment]
Thank for ticket, with hostname-backchannel-dynamic=true it works as previos
